### ISSUE TITLE
would be nice to have the `reply` method available to the custom forbiddenPageFunction

### ISSUE THREAD

**Issue by dperrymorrow** [charlesread/hapi-acl-auth#1]
Thanks for this plugin, it was exactly what i needed for our project to restrict certain users from certain urls.

However, i'm trying to render an error page in this `forbiddenPageFunction` i can render my template just fine using `server.render` but i have no way of setting the status code, so currently my 403 is rendering 200.

```javascript
forbiddenPageFunction(user) {
  return server.render("error", {
    statusCode: 403,
    errorTitle: "Not allowed",
    errorMessage: "You are not allowed to view that page",
  });
},
```
     
I am using vision and would like to be able to would like to be able
```javascript
forbiddenPageFunction(user, reply) {
  return reply.view("error", {
    statusCode: 403,
    errorTitle: "Not allowed",
    errorMessage: "You are not allowed to view that page",
  }).code(403);
},
```
giving the response the proper statusCode

**Comment by charlesread** [comment]
HI, @dperrymorrow !  I'm glad that it is useful for you!  The change that you suggest would be almost trivial to make.  My only concern would be that currently whatever `forbiddenPageFunction()` returns is passed directly to `reply()` (https://github.com/charlesread/hapi-acl-auth/blob/master/index.js#L40), so if you were to `reply` in `forbiddenPageFunction()` the reply interface would be called twice.  Suggestions?

**Comment by charlesread** [comment]
Actually, nevermind.  Added in https://github.com/charlesread/hapi-acl-auth/commit/80011c9a6844c1b14cbeb78d9ab20f9cfc6b9445 , published to NPM as 0.4.0.  `forbiddenPageFunction(user, request, reply)` is now a thing, now `hapi-acl-auth` will only `reply()` if `forbiddenPageFunction()` is passed one argument.  (I'll document it later, but it should work, and is backwards compatible with the previous functionality)

https://github.com/charlesread/hapi-acl-auth/blob/master/index.js#L41

;)

**Comment by dperrymorrow** [comment]
thanks for responding, perhaps I'm not understanding, yes i can return what i want from that function and its passed to reply. Thats all fine, but it renders as a 200.  How can I have a custom function while still having a 403 header. I don't know of another way than having access to the reply function so I can chain off it `reply().code(403)`

**Comment by charlesread** [comment]
So before I made the change `hapi-acl-auth` _always_ passed that which was returned by `forbiddenPageFunction()` to `reply()`, now it still will, but only if your `forbiddenPageFunction()` takes in only a single argument, if it takes in more than one argument `hapi-acl-auth` will basically do nothing.  
```js
forbiddenPageFunction: function(user) {
 return "Sorry, ${user.firstName}, you can't do that!"; 
}
```
Will [still] cause `hapi-acl-auth` to `reply` "Sorry, Charles, you can't do that!" with a 200 (old functionality).  But:
```js
forbiddenPageFunction: function(user, request, reply) {
  reply("Sorry, ${user.firstName}, you can't do that!").code(403);
}
```
Will cause `hapi-acl-auth` to basically do nothing, your forbiddenPageFunction will still be called though, thus giving you the exact functionality that you originally mentioned.

Basically, if your forbiddenPageFunction takes in more that one argument it's up to _you_ to `reply`, whereas if your forbiddenPageFunction only takes in one argument `hapi-acl-auth` _will_ reply for you (just with a 200).

See [this test](https://github.com/charlesread/hapi-acl-auth/blob/master/test/integration.js#L82) for an example.

I threw in the `request` object just for completeness.

**Comment by dperrymorrow** [comment]
👍  so awesome, thanks for your attention on this.

**Comment by charlesread** [comment]
@dperrymorrow , please see the new functionality in the 1.x release, now you'll need something like this:

```js
const response = h.response()
response.code(407)
return reponse.takeover()
```
